src/liberty/libertyProject.ts

```diff
@@ -505,26 +525,29 @@ export class LibertyProject extends vscode.TreeItem {
 }
 
 export async function createProject(context: vscode.ExtensionContext, buildFile: string, contextValue: string, xmlString?: string): Promise<LibertyProject> {
-	let label = "";
-	if (xmlString !== undefined) {
-		// eslint-disable-next-line @typescript-eslint/no-var-requires
-		const parseString = require("xml2js").parseString;
-		parseString(xmlString, (err: any, result: any) => {
-			if (result.project.artifactId[0] !== undefined) {
-				label = result.project.artifactId[0];
-			} else {
-				const dirName = vscodePath.dirname(buildFile);
-				label = vscodePath.basename(dirName);
-			}
-		});
-	} else {
-		label = await gradleUtil.getGradleProjectName(buildFile);
-	}
-	const project: LibertyProject = new LibertyProject(context, label, vscode.TreeItemCollapsibleState.None, buildFile, "start", contextValue, undefined, {
+	const project: LibertyProject = new LibertyProject(context, await getLabelFromBuildFile(buildFile, xmlString), vscode.TreeItemCollapsibleState.None, buildFile, "start", contextValue, undefined, {
```

Comment:
for debugging and code readability purposes, I would recommend taking out the `getLabelFromBuildFile` method call from within the constructor call parameter list:
```
let label = await getLabelFromBuildFile(buildFile, xmlString);
const project: LibertyProject = new LibertyProject(context, label, vscode.TreeItemCollapsibleState.None, buildFile, etc...)
```